common/quick_panel.py

```diff
@@ -2,12 +2,23 @@
 from . import util
 
 
-def show_paginated_panel(items, on_done, limit=6000, **kargs):
+def show_paginated_panel(items, on_done, flags=None, selected_index=None, on_highlight=None,
+                         limit=6000, format_item=None, on_done_item=False, next_message=None):
```

Comment:
Such a high arity seems suboptimal, and suggests that this function and class are doing too much.  Is there any way to defer some of the responsibility to the consumer, and expose the necessary bits there?

One idea would be to return the `PaginatedPanel` from this function, and allow the consumer to act on it within a callback.  Another would be to accept callback functions that are invoked with other functions (e.g. for panel continuation).